### ISSUE TITLE
🚀 Feature: Make users searchable using email

### ISSUE THREAD

**Issue by rdmchr** [appwrite/appwrite#2834]
### 🔖 Feature description

**This feature is meant for the server SDKs.**
I think it's an important feature to make users searchable using their email addresses. Something like
```JS
const users = new sdk.Users(appwrite);
users.search('hello@appwrite.io');
```
would be really useful.

A related feature would be to make users searchable using their preferences or name, but I think those wouldn't be as useful as this.

### 🎤 Pitch

Let's say I'm making an invitation system for users. Currently, I have to create a collection containing all user ids + their email addresses. This collection has to always be kept up to date when let's say a user changes their email address.
If my idea would be implemented I could just query the user using their email address with information already in the database.
This would of course scale better since I won't need to save the same information twice, but also be more reliable (see example above).

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
There is a search parameter in the list users api call: https://appwrite.io/docs/server/users#usersList

would that work for you?

**Comment by rdmchr** [comment]
I think? How exactly would I write a search query to check for email or name?

**Comment by stnguyen90** [comment]
Considering [this](https://github.com/appwrite/sdk-for-node/blob/master/lib/services/users.js#L21) is the function signature, possibly:

```javascript
const users = new sdk.Users(appwrite);
users.list('hello@appwrite.io');
```

**Comment by stnguyen90** [comment]
Were you able to try the search?

**Comment by rdmchr** [comment]
Yeah, it kind of works. The syntax is this:
```TS
const users = new sdk.Users(appwrite);
console.log(await users.list(Query.equal('email', 'hello@appwrite.io')));
```
But the problem is that the search is not really "exact" enough if you can call it that. Lets say I have two users: "hello@appwrite.io" and "he@appwrite.io". If I search for "hello@appwrite.io" I get only that user, but if I search for "he@appwrite.io" I get both users.

**Comment by stnguyen90** [comment]
You should definitely only be passing the search term rather than `Query.equal('email', 'hello@appwrite.io')`. However, I do see that pass `"hello@appwrite.io"` causes a 500 error. This sounds related to https://github.com/appwrite/appwrite/issues/2394 since the search mode has been changed back to boolean mode.

For now, the workaround is to wrap in double quotes, like `'""hello@appwrite.io""'`, however, it would be nice if Appwrite handled this automatically.

**Comment by stnguyen90** [comment]
Related PR: https://github.com/utopia-php/database/pull/113.

After this PR is merged, I would assume this would work:

```javascript
const users = new sdk.Users(appwrite);
console.log(await users.list('hello@appwrite.io');
```

FYI, searching for `he@appwrite.io` might be returning both because words less than 3 characters are ignored so it's actually only searching for `appwrite`.

**Comment by TorstenDittmann** [comment]
0.13.3 got released and this should be fixed now 👏🏻 

One downside of the current approach is that you would need to wrap the email in quotes. Reference => https://github.com/appwrite/appwrite/issues/2394#issuecomment-1067882013

**Comment by TorstenDittmann** [comment]
Actually I am gonna re-open this issue again - until we allow proper filtering by an email 👍🏻

**Comment by stnguyen90** [comment]
As of 0.15.2, searching on an email may return multiple results (especially if there are multiple domain matches). 

<img width="984" alt="image" src="https://user-images.githubusercontent.com/1477010/180869426-35c3b189-adcb-4c22-a61f-dd17b8f9e494.png">

To get more of an exact match, you can wrap your search term in double quotes:

<img width="977" alt="image" src="https://user-images.githubusercontent.com/1477010/180869664-2403682e-8a75-4444-9b83-38b927628691.png">

@TorstenDittmann, we good to close this?

**Comment by TorstenDittmann** [comment]
This got implemented and released with Appwrite 1.0.0 🥳  You can now Query directly by the E-Mail.

Thank you for your contribution 🙏🏻  You can read more [here](https://appwrite.io/1.0).

**Comment by AneesRazaRaise** [comment]
After four days of effort and extensive research, finally came to the solution. The correct syntax of searching user by there 'email' in node.js is: 
`users.list([Query.equal("email","m.anees7714@gmail.com")]);`

and the complete code would be: 

`const {Client, Query, Users} = require("node-appwrite");`


`const _client = new Client();`

`_client
      .setEndpoint(req.variables['APPWRITE_FUNCTION_ENDPOINT'])`
      `.setProject(req.variables['APPWRITE_FUNCTION_PROJECT_ID'])`
      `.setKey(req.variables['APPWRITE_FUNCTION_API_KEY'])`
      `.setSelfSigned(true);`

`const users = new Users(_client);`

`const promise = users.list([Query.equal("email","m.anees7714@gmail.com")]);`

`promise.then(function (response) {`
`    console.log(response);`
`}, function (error) {`
`    console.log(error);`
`});`
